app/assets/javascripts/discourse/app/mixins/textarea-text-manipulation.js

```diff
@@ -272,6 +273,19 @@ export default Mixin.create({
       }
     }
 
+    if (plainText && !handled && selected.end > selected.start) {
+      let isURL;
+      try {
+        isURL = !!new URL(plainText);
```

Comment:
Let me know if folks have a preferred way of testing for links in Discourse. Generally it's link-like enough if the URL constructor is able to parse the string.

https://url.spec.whatwg.org/#valid-url-string

Reply:
This seems good to me, but I do wonder if we will end up linking things that are not links, and if that will be more of a nuissance than the thing the auto linking solves?

Reply:
> I do wonder if we will end up linking things that are not links

In practice, I haven't seen a major report like this yet for folks using the block or classic editor in WordPress. We do get small issues from time to time, but they tend to be minor inconveniences.

That said, I browsed around the codebase some more and it does look like discourse-markdown uses https://github.com/markdown-it/linkify-it to test links. If we'd like this pasting behavior, ideally we can match the rules here. I'll take a look to see if it's accessible to d-editor.

Example of a cooked link in preview:
https://user-images.githubusercontent.com/1270189/142941831-7e07a6b5-8d10-4bb1-98c1-b723883f9c24.mp4

Another option might be to limit this to a site or user setting to toggle the behavior.

Reply:
I sketched out an approach in 5a9cf8d where we reuse the linkify rules. Not sure if I like using library implementation details in this way, but I do feel like the overall behavior is much improved. One alternative might be using linkify directly versus off of the PrettyText instance.

 I'll add some example tests/videos tomorrow, but let me know if you had feedback/questions ahead of time.